engine/table/src/test/java/io/deephaven/engine/table/impl/QueryTableFloatingPointZeroTest.java

```diff
@@ -0,0 +1,599 @@
+    //
+    // }
+
+    @Test
+    public void testFloatAggKey() {
+        {
+            final Table xy = newTable(floatCol("X", -0.0f, 0.0f), intCol("Y", 0, 1));
+            final Table grouped = xy.groupBy("X");
+            floatToBitsEquals(grouped, "X", oneKey(grouped), -0.0f);
+        }
+        {
+            final Table xy = newTable(floatCol("X", 0.0f, -0.0f), intCol("Y", 0, 1));
+            final Table grouped = xy.groupBy("X");
+            floatToBitsEquals(grouped, "X", oneKey(grouped), 0.0f);
+        }
+    }
+
+    @Test
+    public void testDoubleAggKey() {
+        {
+            final Table xy = newTable(doubleCol("X", -0.0, 0.0), intCol("Y", 0, 1));
+            final Table grouped = xy.groupBy("X");
+            doubleToBitsEquals(grouped, "X", oneKey(grouped), -0.0);
+        }
+        {
+            final Table xy = newTable(doubleCol("X", 0.0, -0.0), intCol("Y", 0, 1));
+            final Table grouped = xy.groupBy("X");
+            doubleToBitsEquals(grouped, "X", oneKey(grouped), 0.0);
+        }
+    }
+
+    @Test
+    public void testFloatMinMaxAggAndFormulaResult() {
+        {
+            final Table negFirst = newTable(floatCol("X", -0.0f, 0.0f));
+            for (Table table : Arrays.asList(
+                    negFirst.minBy(),
+                    negFirst.maxBy(),
+                    negFirst.groupBy().view("X=min(X)"),
+                    negFirst.groupBy().view("X=max(X)"))) {
+                floatToBitsEquals(table, "X", oneKey(table), -0.0f);
+            }
+        }
+        {
+            final Table posFirst = newTable(floatCol("X", 0.0f, -0.0f));
+            for (Table table : Arrays.asList(
+                    posFirst.minBy(),
+                    posFirst.maxBy(),
+                    posFirst.groupBy().view("X=min(X)"),
+                    posFirst.groupBy().view("X=max(X)"))) {
+                floatToBitsEquals(table, "X", oneKey(table), 0.0f);
+            }
+        }
+    }
+
+    @Test
+    public void testDoubleMinMaxAggAndFormulaResult() {
+        {
+            final Table negFirst = newTable(doubleCol("X", -0.0, 0.0));
+            for (Table table : Arrays.asList(
+                    negFirst.minBy(),
+                    negFirst.maxBy(),
+                    negFirst.groupBy().view("X=min(X)"),
+                    negFirst.groupBy().view("X=max(X)"))) {
+                doubleToBitsEquals(table, "X", oneKey(table), -0.0);
+            }
+        }
+        {
+            final Table posFirst = newTable(doubleCol("X", 0.0, -0.0));
+            for (Table table : Arrays.asList(
+                    posFirst.minBy(),
+                    posFirst.maxBy(),
+                    posFirst.groupBy().view("X=min(X)"),
+                    posFirst.groupBy().view("X=max(X)"))) {
+                doubleToBitsEquals(table, "X", oneKey(table), 0.0);
+            }
+        }
+    }
+
+    @Test
+    @Ignore("Unstable sort?")
+    public void testFloatMedianAggResult() {
```

Comment:
This is probably less well-defined than we'd hope. I bet that the SSM bucket gets created for -0.0 or 0.0, whichever is encountered first, and we'll keep using that "forever".